### ISSUE TITLE
Intelligent Routing Keys

### ISSUE THREAD

**Issue by timolegros** [hicommonwealth/commonwealth#8369]
## Description
We currently route all `ThreadCreated` and `ThreadUpvoted` events to the CommonwealthConsumer even if the thread or upvote is not related to a contest. This is causing a huge amount of errors (and retries) that are entirely avoidable if we only route messages to the consumer when the content is related to a contest.

### Tasks
- Create a function which builds a RabbitMQ routing key based off event metadata and call this function from the broker `publish` method`.
- Update bindings to adhere to the new routing keys.

## Additional context
Follow-up to #8366

**Comment by timolegros** [comment]
Blocked until #8366 is merged.

**Comment by timolegros** [comment]
I can't test these changes on Frick because contests launching is broken on Frick.

First I had to deal with a bug where if you have Phantom _and_ MetaMask enabled you get a cryptic `User rejected the request` error because Phantom mocks MetaMask (disabled Phantom after digging through source files).

After that was fixed I managed to deploy stakes but now I can't deploy contests:


![Image](https://github.com/user-attachments/assets/9319d946-7c7b-4b13-bafc-27b189162836)

I can't move forward until this is resolved and with both Marcin and Malik out and unable to help fix this client-side bug this ticket is officially blocked again.
